### PR TITLE
Add dostrip for EAPI 7

### PR DESCRIPTION
**Pull request by zmedico**
This patch includes the essential parts of the dostrip implementation
from portage-mgorny. All of the prepstrip code has moved to
bin/estrip, without any changes except the addition of argument
parsing for estrip --ignore, --queue, and --deque modes which are
equivalent to the corresponding ecompressdir modes.

Due to overlap, also ban the non-standard prepstrip and prepallstrip
helpers in EAPI 7, with a die message suggesting to use 'dostrip'
instead. Also ignore the non-standard STRIP_MASK variable for EAPI 7.

Bug: https://bugs.gentoo.org/203891

### CODE REVIEW

File: bin/ebuild-helpers/prepall [hunk 1]
```diff
@@ -19,7 +19,7 @@ fi
 prepallman
```
Comment (mgorny) [inline bin/ebuild-helpers/prepall]: On a related note, you probably want to ban all of `prep*` in EAPI 7.

File: bin/estrip [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+else
+	get_inode_number() { stat -c '%i' "$1"; }
+fi
+cd "${tmpdir}/inodes" || die "cd failed unexpectedly"
+while read -r x ; do
+	inode_link=$(get_inode_number "${x}") || die "stat failed unexpectedly"
+	echo "${x}" >> "${inode_link}" || die "echo failed unexpectedly"
+done < <(
+	# Use sort -u to eliminate duplicates for bug #445336.
+	(
```
Comment (mgorny) [inline bin/estrip]: But now you're not using the results of `--queue` and instead stripping all files again…

Comment (zmedico) [reply bin/estrip]: Thanks, updated to use this code conditionally:
```bash
while IFS= read -d '' -r x ; do
	rm -f "${x}" || die
	inode_link=$(get_inode_number "${x%.estrip}") || die "stat failed unexpectedly"
	echo "${x%.estrip}" >> "${inode_link}" || die "echo failed unexpectedly"
done < <(find "${ED}" -name '*.estrip' -print0)
```

File: bin/ebuild-helpers/prepall (hunk 1 above)
Comment (zmedico) [reply bin/ebuild-helpers/prepall]: Sure, we can do that in a separate patch. I this patch, I'm only concerned with *strip commands due to overlap.

File: bin/estrip [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+		banner=true
+	fi
+
+	(
+	__multijob_child_init
+	f=$(file "${x}") || exit 0
+	[[ -z ${f} ]] && exit 0
+
+	if ! ${prepstrip}; then
+		strip_this=true
```
Comment (zmedico) [inline bin/estrip]: Updated to ignore STRIP_MASK here for EAPI 7.

Comment (zmedico) [reply bin/estrip]: For clarity, I've split the changes into a separate commit from the prepstrip -> bin/estrip rename.

**Pull request merged**